integrated_channels/integrated_channel/learner_data.py

```diff
@@ -132,12 +132,14 @@ def collect_learner_data(self):
             else:
                 completed_date, grade, is_passing = self._collect_grades_data(enterprise_enrollment, course_details)
 
-            yield self.get_learner_data_record(
+            record = self.get_learner_data_record(
                 enterprise_enrollment=enterprise_enrollment,
                 completed_date=completed_date,
                 grade=grade,
                 is_passing=is_passing,
             )
+            if record:
```

Comment:
Nit: Might be useful to add a comment here explaining to future developers why we need to differentiate between when a record is returned and when it isn't.